core/src/main/java/org/infinispan/util/concurrent/locks/KeyAwareLockListener.java

```diff
@@ -0,0 +1,21 @@
+package org.infinispan.util.concurrent.locks;
+
+/**
+ * A listener for {@link KeyAwareLockPromise}.
+ * <p>
+ * This event contains the key that the lock owner is try to acquire.
+ *
+ * @author Pedro Ruivo
+ * @since 8.0
+ */
+public interface KeyAwareLockListener {
```

Comment:
yes. extended-stats modules needs to have some association between keys and locks.